client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -310,9 +323,39 @@ class CancelPurchaseForm extends React.Component {
 		const reasons = {};
 		const { translate } = this.props;
 		const { questionOneOrder, questionOneRadio, questionOneText } = this.state;
+		const { productSlug: productBeingRemoved } = this.props.purchase;
+
+		// get all downgradable plans and products for downgrade question dropdown
+		const downgradableJetpackPlansAndProducts = [
+			...DOWNGRADEABLE_PLANS_FROM_PLAN[ productBeingRemoved ],
+			...JETPACK_PRODUCTS_LIST,
+		];
+		const selectBoxItems = downgradableJetpackPlansAndProducts
+			.map( slugToSelectorProduct )
+			// filter out any null items
+			.filter( ( product ) => product )
+			// get only annual products/plans (no need for monthly variants in the dropdown)
+			.filter( ( product ) => product.term === TERM_ANNUALLY )
+			.map( ( product ) => ( {
+				value: product.productSlug,
+				label: translate( 'Jetpack {{planName/}}', {
+					components: { planName: <>{ product.shortName }</> },
+				} ),
+			} ) );
+
+		const dropDownSelectOptions = [
+			{ value: 'select_a_product', label: translate( 'Select a product' ), isLabel: true },
+			...selectBoxItems,
+		];
+
+		const initialSelected = downgradableJetpackPlansAndProducts.includes(
+			this.state.questionOneText
+		)
+			? this.state.questionOneText
+			: 'select_a_product';
 
 		const appendRadioOption = ( key, radioPrompt, textPlaceholder ) =>
-			( reasons[ key ] = radioOption(
+			( reasons[ key ] = radioTextOption(
```

Comment:
I know it comes from the previous implementation, but I'm a bit confused about why `radioTextOption` is used like a regular function here, since it's a React component.

Reply:
Yes, TBH I was a little confused about this also. Since there's quite a lot going on in this file, and to avoid introducing any new issues, I just kinda followed along with how the previous implementation was doing it. 😉